### PR TITLE
Alfoa/failed runs

### LINKED ISSUE #268: Repeat failed model runs
--------
Issue Description
--------
##### What did you expect to see happen?
It is possible that a external model or code crashes for a certain constellation of inputs. If that happens, RAVEN should repeat that run, i.e. try again.

##### What did you see instead?
RAVEN quits.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or improvement? ***improvement***
- [x] 2. Is it tagged with a priority: critical, normal or minor? ***minor***
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements? ***n/a***
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users. ***no***
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.) ***yes***

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest stable branch? If yes, is there any issue tagged with stable (create if needed)?
- [x] 5. If the issue is being closed without a merge request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by alfoa**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)

Closes #268 

##### What are the significant changes in functionality due to this change request?
This adds the possibility to rerun a certain evaluation multiple times in case it fails.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

**Comment by alfoa** [comment]
anyone??

File: doc/user_manual/step.tex [hunk 1]
```diff
@@ -103,6 +103,9 @@ \subsection{SingleRun}
 in the \xmlNode{RunInfo} block, under the \xmlNode{Sequence} node. If the name
 of this \textbf{Step} is not listed in the  \xmlNode{Sequence} block, its action is not
 going to be performed.
+\item \xmlAttr{repeatFailureRuns}, \xmlDesc{optional integer attribute}, this optional 
+attribute could be used to set a certain number of repeations that need to be performed
```
Comment (PaulTalbot-INL) [inline doc/user_manual/step.tex]: "repeations" -> "repititions"

File: doc/user_manual/step.tex [hunk 2]
```diff
@@ -215,6 +218,15 @@ \subsection{MultiRun}
 in the \xmlNode{RunInfo} block, under the \xmlNode{Sequence} node. If the name
 of this \textbf{Step} is not listed in the  \xmlNode{Sequence} block, its action is not
 going to be performed.
+\item \xmlAttr{re-seeding}, \xmlDesc{optional integer/string attribute}, this optional 
+attribute could be used to control the seeding of the random number generator (RNG). 
+If inputted, the RNG can be reseeded. The value of this attribute
+can be: either 1) an integer value with the seed to be used (e.g. \xmlAttr{re-seeding} = 
+``20021986''), or 2) string value named ``continue'' where the RNG is not re-initialized
+\item \xmlAttr{repeatFailureRuns}, \xmlDesc{optional integer attribute}, this optional 
+attribute could be used to set a certain number of repeations that need to be performed
```
Comment (PaulTalbot-INL) [inline doc/user_manual/step.tex]: "repeations"

File: doc/user_manual/step.tex (hunk 1 above)
Comment (alfoa) [inline doc/user_manual/step.tex]: I think you mean "repetitions".
"repeations" -> "repititions" ->  "repetitions"

**Comment by alfoa** [comment]
Addressed @PaulTalbot-INL 's comments

**Pull request merged**